src/WooCommerceServiceProvider.php

```diff
@@ -42,10 +42,12 @@ public function bindFilters()
     {
         $woocommerce = $this->app['woocommerce'];
 
-        add_filter('template_include', [$woocommerce, 'templateInclude'], 11);
-        add_filter('woocommerce_locate_template', [$woocommerce, 'template']);
-        add_filter('wc_get_template_part', [$woocommerce, 'template']);
-        add_filter('comments_template', [$woocommerce, 'reviewsTemplate'], 11);
+        add_filter('template_include', [$woocommerce, 'templateInclude'], 100, 1);
+        add_filter('woocommerce_locate_template', [$woocommerce, 'templatePart'], PHP_INT_MAX);
+        add_filter('wc_get_template_part', [$woocommerce, 'templatePart'], PHP_INT_MAX);
```

Comment:
Why would we need `PHP_INT_MAX` here? Did you encounter an issue with some plugin?

Reply:
Yes, this hook should be the last of all, and this is important.

Reply:
Many programmers use this value as well ```(PHP_INT_MAX-1)``` to give another chance for the final change.

Reply:
Yeah I've on occasion encountered plugins that used it and required me to unhook their filter so i can add it with a lower priority :)

I don't really see a reason for having it higher than it needs to be tough. Here you can see what roots/acorn uses https://github.com/roots/acorn/blob/master/src/Acorn/Sage/SageServiceProvider.php#L48:L85